src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -211,7 +217,23 @@ CreateModifyPlan(Query *originalQuery, Query *query,
 		return multiPlan;
 	}
 
-	ereport(DEBUG2, (errmsg("Creating router plan")));
+	if (*multiShardModifyQuery)
+	{
+		/* although the plan is a router one, we still touch multiple shards */
+		if (MultiTaskQueryLogLevel != MULTI_TASK_QUERY_INFO_OFF)
+		{
+			ereport(MultiTaskQueryLogLevel, (errmsg(
```

Comment:
it would be nice if we could keep these checks in one place. For real-time SELECT it's in `JobExecutorType`, though that seems a little weird as well. Maybe move both to `FinalizePlan`?